Update presence types & documentation

SethFalco opened the pull request:
<!-- Thanks for submitting a PR! Please ensure the following requirements are met in order for us to review your PR -->

## Checklist

* [x] Tests written for all new code
* [x] Linter has been satisfied
* [x] Sign-off given on the changes (see CONTRIBUTING.md)

While working on a PR for matrix-appservice-discord, I was tripped up by the presence states.
I usually only look in the JSDocs for information while working on projects/PRs, and the term **unavailable** wasn't clearly defined or linked to before.

Looked to me that `"unavailable"` was either:
* Home server has presence disabled, so the presence is unavailable.
* DND

I had to ask in chat to find out it was actually neither of those things. ^-^'

Maybe for convenience for myself and others, could we just add a link to the spec that clearly defines these?

turt2live commented:
Unfortunately the spec isn't the most helpful resource for who this SDK is trying to target: beginner or near-beginner bot writers (obviously still supporting the more advanced users too). The SDK is far from that point currently, but I'd like to avoid spec links for now.

If there's a guide which can be linked to (or created), that would be preferable

SethFalco commented:
> Unfortunately the spec isn't the most helpful resource for who this SDK is trying to target

Tbh, I thought the spec did a pretty good job in this case at least. Though I'll admit the spec is a bit intimidating to check out, it did a good job.

However, if we're keen on not putting spec links for now, I could try and find a guide, or just duplicate some of the information over as JSDocs.

turt2live commented:
A guide or duplicating would be fine, yea

SethFalco commented:
I couldn't find a guide for it, so opted to duplication.
Probably depends on the processor, but VSC renders Markdown in JSDocs so just did it in Markdown.

Also added the `@see` in `PresenceEventContent#presence` since it's more often going to be exposed through this, so it just directs users to where they can see the breakdown.

![image](https://user-images.githubusercontent.com/22801583/194777966-a19fda8c-7152-47e8-9934-cc8bf53b6d80.png)

![image](https://user-images.githubusercontent.com/22801583/194777980-26446502-8ebd-4389-8cfa-2595d2001b15.png)

Turns out this was actually documented in `MatrixClient#syncingPresence` I noticed, but I never ran into that while consuming the SDK while working with matrix-appservice-discord. I still think keeping it in `PresenceState` is more appropriate, but haven't removed it where it was already documented.

---

On the side, I noticed the type strings `"online" | "offline" | "unavailable"` were done in multiple places. 🤔 
Is there any reason we couldn't just have `PresenceState` used in those places? (I updated it in this PR, but please let me know if that should be undone.)

turt2live reviewed (approved):
Thanks :)
